### ISSUE TITLE
Plugin does not clean output directory in watch mode

### ISSUE THREAD

**Issue by github-account-because-they-want-it** [johnagan/clean-webpack-plugin#80]
Steps to reproduce:

1. Clone my repo [here](https://github.com/github-account-because-they-want-it/webpack-struggle.git)
2. `npm install`
3. Run `webpack --watch` in the project dir. The output is generated as expected:
```
ls -l dist
index.html  main.e4ff9438d89750f15374.js  runtime~main.297929bbe0221ed990a6.js
 ```
4. Modify *main.js* and save. The output is rebuilt, with 4 files instead of 3:
```
ls -l dist
index.html                    main.e4ff9438d89750f15374.js
main.64d950fb1144a8614c58.js  runtime~main.297929bbe0221ed990a6.js
```
If you repeat step 4, you get 5 files, then 6...

Is there something wrong with my build? Isn't this a supported use case?

**Comment by octaharon** [comment]
It's perfectly fine, since watch mode compiles everything to the memory, there's nothing to clear. You can see this behavior is intended here:
https://github.com/johnagan/clean-webpack-plugin/blob/master/index.js#L199

**Comment by axefrog** [comment]
Aren't you confusing watch mode with webpack-dev-server? The latter compiles to memory so it can serve changes after recompilation, but regular watch mode is all about regenerating the output folder when files change. Compiling it only to memory would be pointless.

**Comment by chrisdothtml** [comment]
Same problem here. Even if I provide `watch: true`, it still doesn't work in watch mode

**Comment by tedbarth** [comment]
It does not feel like a good usability if you have to set `{ watch = true }` first to enable cleaning when files changed while having webpack started with `--watch` flag. A lot more people will hit the situation we hit here.

I suggest you change the default to `watch = true`.

**Comment by johnagan** [comment]
>Not sure why this wouldn't be enabled by default, or why it's even an option in the first place though

I tend to agree here... Let me see about changing it to default.

**Comment by chrisblossom** [comment]
Closed in [`v2.0.0`](https://github.com/johnagan/clean-webpack-plugin/tree/v2.0.0) via #99. See [johnagan/clean-webpack-plugin#usage](https://github.com/johnagan/clean-webpack-plugin#usage). Please open a new issue if you continue to have problems.
